src/bspline_curve.rs

```diff
@@ -0,0 +1,80 @@
+use cgmath::{Vector2, Vector3, Vector4};
+
+use buffer::Buffer;
+use device::Device;
+use geometry::Geometry;
+use sys::*;
+use {BufferType, Format, GeometryType, CurveType};
+
+pub struct BsplineCurve<'a> {
+    device: &'a Device,
+    pub(crate) handle: RTCGeometry,
+    pub vertex_buffer: Buffer<'a, Vector4<f32>>,
+    pub index_buffer: Buffer<'a, u32>,
+    pub normal_buffer: Buffer<'a, Vector3<f32>>,
```

Comment:
We'll want to make this a `pub normal_buffer: Option<Buffer<'a, Vector3<f32>>>` too so we don't have to make a new buffer each type which might be unused (as down on line 27 here). So if the caller isn't using a normal buffer, no memory for one is allocated at all